hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/table/HoodieTableFactory.java

```diff
@@ -86,12 +84,14 @@ public DynamicTableSource createDynamicTableSource(Context context) {
     setupTableOptions(conf.getString(FlinkOptions.PATH), conf);
     ResolvedSchema schema = context.getCatalogTable().getResolvedSchema();
     setupConfOptions(conf, context.getObjectIdentifier(), context.getCatalogTable(), schema);
-    return new HoodieTableSource(
+    HoodieTableSource source = new HoodieTableSource(
         schema,
         path,
         context.getCatalogTable().getPartitionKeys(),
         conf.getString(FlinkOptions.PARTITION_DEFAULT_NAME),
```

Comment:
Let's keep the sanity check in the `HoodieTableFactory`?

Reply:
If according to what you said,  Metadataclient needs to be initialized in the factory(Hoodie source sanity check need). It seems more reasonable to initialize in source?

Reply:
>  Metadataclient needs to be initialized in the factory

That's okay, we already do that for the sink sanity check of table config.

Reply:
@danny0405  cc

Reply:
@danny0405  cc